core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -876,8 +889,18 @@ class DAGScheduler(
       // SparkListenerStageCompleted here in case there are no tasks to run.
       outputCommitCoordinator.stageEnd(stage.id)
       listenerBus.post(SparkListenerStageCompleted(stage.latestInfo))
-      logDebug("Stage " + stage + " is actually done; %b %d %d".format(
-        stage.isAvailable, stage.numAvailableOutputs, stage.numPartitions))
+
+      val debugString = stage match {
+        case stage: ShuffleMapStage =>
+          "Stage " + stage + (" is actually done; " +
+            "(available: ${stage.isAvailable}," +
```

Comment:
you need to be an s before the strings to tell scala to do string interpolation (e.g., s"(available: ..."), otherwise the resulting string will print the $ etc. rather than printing the value.